web-prover-core/src/parser/config.rs

```diff
@@ -22,11 +38,15 @@ pub struct ExtractorConfig {
 }
 
 impl ExtractorConfig {
-  /// Extracts and validates data using this extractor configuration
-  pub fn extract_and_validate(&self, data: &Value) -> Result<ExtractionResult, ExtractorError> {
-    match self.format {
-      DataFormat::Json => extractor::extract_json(data, self),
-    }
+  /// Use document-specific extractors to extract and validate data
+  pub fn extract_and_validate(&self, data: &[u8]) -> Result<ExtractionResult, ExtractorError> {
+    let extractor: Box<dyn DocumentExtractor> = match self.format {
+      DataFormat::Json => Box::new(JsonDocumentExtractor),
+      DataFormat::Html => Box::new(HtmlDocumentExtractor),
+    };
```

Comment:
```suggestion
    let extractor: impl DocumentExtractor = match self.format {
      DataFormat::Json => JsonDocumentExtractor,
      DataFormat::Html => HtmlDocumentExtractor,
    };
```

does this work? Ever since return position impl trait was stabilized, I think we don't need to `Box<dyn Trait>` as much.

Reply:
You might need an `as impl DocumentExtractor` or something in there, but i'm 99% sure this is doable.

Reply:
I may be missing something but I wasn't able to do that without spilling generics everywhere